Name Owhonda Ihunwo

Labpartner(s)

In [16]:
# import statements go here
import pandas as pd
import numpy as np

# Class 13.2

# Warmups 13.2

**W.1 Pandas Practice** 

Create a pandas series from each of the items below: a list, numpy and a dictionary

mylist = list('abcedfghijklmnopqrstuvwxyz')

myarr = np.arange(26)

mydict = dict(zip(mylist, myarr))

In [4]:
mylist = list('abcedfghijklmnopqrstuvwxyz')
myarr = np.arange(26)
mydict = dict(zip(mylist, myarr))

series_list = pd.Series(mylist)
print("Series from list:\n", series_list)

series_arr = pd.Series(myarr)
print("\nSeries from numpy array:\n", series_arr)

series_dict = pd.Series(mydict)
print("\nSeries from dictionary:\n", series_dict)

Series from list:
 0     a
1     b
2     c
3     e
4     d
5     f
6     g
7     h
8     i
9     j
10    k
11    l
12    m
13    n
14    o
15    p
16    q
17    r
18    s
19    t
20    u
21    v
22    w
23    x
24    y
25    z
dtype: object

Series from numpy array:
 0      0
1      1
2      2
3      3
4      4
5      5
6      6
7      7
8      8
9      9
10    10
11    11
12    12
13    13
14    14
15    15
16    16
17    17
18    18
19    19
20    20
21    21
22    22
23    23
24    24
25    25
dtype: int32

Series from dictionary:
 a     0
b     1
c     2
e     3
d     4
f     5
g     6
h     7
i     8
j     9
k    10
l    11
m    12
n    13
o    14
p    15
q    16
r    17
s    18
t    19
u    20
v    21
w    22
x    23
y    24
z    25
dtype: int32


**W.2 Pandas Practice con** 

Convert the series you just created into a dataframe with its index as another column on the dataframe.

In [10]:

df_list = series_list.to_frame('values')
df_list['index'] = df_list.index
print("Dataframe from series_list:\n", df_list)


df_arr = series_arr.to_frame('values')
df_arr['index'] = df_arr.index
print("\nDataframe from series_arr:\n", df_arr)


df_dict = series_dict.to_frame('values')
df_dict['index'] = df_dict.index
print("\nDataframe from series_dict:\n", df_dict)

Dataframe from series_list:
    values  index
0       a      0
1       b      1
2       c      2
3       e      3
4       d      4
5       f      5
6       g      6
7       h      7
8       i      8
9       j      9
10      k     10
11      l     11
12      m     12
13      n     13
14      o     14
15      p     15
16      q     16
17      r     17
18      s     18
19      t     19
20      u     20
21      v     21
22      w     22
23      x     23
24      y     24
25      z     25

Dataframe from series_arr:
     values  index
0        0      0
1        1      1
2        2      2
3        3      3
4        4      4
5        5      5
6        6      6
7        7      7
8        8      8
9        9      9
10      10     10
11      11     11
12      12     12
13      13     13
14      14     14
15      15     15
16      16     16
17      17     17
18      18     18
19      19     19
20      20     20
21      21     21
22      22     22
23      23     23
24      24     24
25      25     

**W.3 Pandas Practice con** 

Combine two series to form a dataframe.

In [12]:
df = pd.concat([series_list, series_arr], axis=1)
print(df)

    0   1
0   a   0
1   b   1
2   c   2
3   e   3
4   d   4
5   f   5
6   g   6
7   h   7
8   i   8
9   j   9
10  k  10
11  l  11
12  m  12
13  n  13
14  o  14
15  p  15
16  q  16
17  r  17
18  s  18
19  t  19
20  u  20
21  v  21
22  w  22
23  x  23
24  y  24
25  z  25


**W.4 Pandas Practice con**

Compute the minimum, 25th percentile, median, 75th, and maximum of the following series:

ser = pd.Series(np.random.normal(10, 5, 25))

What kind of distribution is this?


In [20]:
ser = pd.Series(np.random.normal(10, 5, 25))
print("Minimum: ", ser.min())
print("25th percentile: ", ser.quantile(0.25))
print("Median: ", ser.median())
print("75th percentile: ", ser.quantile(0.75))
print("Maximum: ", ser.max())

Minimum:  -2.8509875574686596
25th percentile:  7.710531010334133
Median:  10.587653991389232
75th percentile:  13.126295984526404
Maximum:  18.64224654003369


# Lecture 13.1

## Agenda:
- Machine Learning in Scikit learn Ch 3-4

# Lab 13.1

#### E.0 Complete the previous labs if you are behind.

#### E.1 Complete Ch 3-4 of "Supervised Learning with scikit-learn". Don't forget to turn in your scores

**Notes** Make notes for yourself on progamming tecniques and commands you learned in the this week, including examples, comments and explainitory text. You can do this here or in a separate notebook that you link to here. Basically, you are making a cheat sheet for yourself.

#### E.2 Work on your final project